app/controllers/admin/running_results_controller.rb

```diff
@@ -0,0 +1,33 @@
+module Admin
+  class RunningResultsController < BaseController
+
+    RUNNING_RESULTS = <<-SQL
+      SELECT
+        title query_title,
+        compiled_body query_body,
+        TO_CHAR(results.started_at, 'YYYY-MM-DD HH24:MI:SS') started_at,
+        ROUND(EXTRACT(EPOCH FROM (now() - results.started_at)::INTERVAL)) duration_seconds,
+        users.name author,
+        users.role author_role,
+        queries.id query_id,
+        query_versions.id AS query_version_id,
+        version
+      FROM results
+        LEFT OUTER JOIN query_versions ON query_versions.id = results.query_version_id
+        LEFT OUTER JOIN queries ON query_versions.query_id = queries.id
+        INNER JOIN users ON results.owner_id = users.id
```

Comment:
Have we confirmed that this join order worked, I was always under the impression that inner joins have to happen first, though in this case it might make no difference due to what tables are being linked.

Reply:
@dtboctor i don't think that's an actual constraint, but you might be thinking about how mysql will not do forward-lookups for tables who have not been joined yet--reading top-to-bottom, left-to-right--in the join clause?

```
mysql> create table a (id int);
Query OK, 0 rows affected (0.22 sec)

mysql> create table b (id int, a_id int);
Query OK, 0 rows affected (0.07 sec)

mysql> create table c (iid int, b_id int);
Query OK, 0 rows affected (0.07 sec)

mysql> select * from a join c on b.id = c.b_id join b on a.id = b.a_id;
ERROR 1054 (42S22): Unknown column 'b.id' in 'on clause'

mysql> select * from a join b on a.id = b.a_id join c on b.id = c.b_id;
Empty set (0.02 sec)
```

this reveals an absurd hack in them mysql parser (how tf are you supposed to trust the query optimizer if the parser can't do lookaheads?), IMO

Reply:
huh yea i've never heard of join ordering mattering. what marc pointed out is weird tho.